keyvi/include/keyvi/dictionary/completion/multiword_completion.h

```diff
@@ -150,12 +150,12 @@ class MultiWordCompletion final {
             data->traverser++;
           } else {
             TRACE("StateTraverser exhausted.");
-            return Match();
+            return match_t();
           }
         }
       };
 
-      return MatchIterator::MakeIteratorPair(tfunc, first_match);
+      return MatchIterator::MakeIteratorPair(tfunc, std::move(first_match));
```

Comment:
Not sure if there is a need to do `std::move`, I believe the `shared_ptr` object is not heavy itself.

Reply:
Some research:

- https://stackoverflow.com/questions/41871115/why-would-i-stdmove-an-stdshared-ptr
- https://stackoverflow.com/questions/77786702/how-to-avoid-clang-tidy-warning-when-passing-shared-ptr-to-thread-function


Sounds like `move` should be used.

The 2nd link mentions `clang-tidy`, it is on my todo list to integrate it into our style check.

Reply:
Cool, thanks! ❤️